css/how-we-write-css.md

```diff
@@ -25,19 +25,22 @@ Contains all **global** SASS variables for your project e.g. colors, fonts, medi
 Contains all **global** SASS functions e.g. em calculation, unit stripping.
 
 #### Mixins
-Contains all **global** SASS mixins, this includes mixin definitions used in the **pattern** and **utility** levels.
+Contains all **global** SASS mixins, this includes mixin definitions used in the **component** and **utility** levels.
 
 #### Base
 A Base rule is applied to an element using an element/type selector, a descendant selector, or a child selector, along with any pseudo-classes. Base styles are related to the basic styles of your product, like typography, reset and global elements like links. Use this level to include any (third-party) resets or normalization css.
 
 #### Components
-Components use classes to map to specific UI elements. They should exist as stand-alone UI components and **never** depend on other components.
+Components use classes to map to specific UI elements.
 
-#### Patterns
-Patterns use classes to apply a reusable design pattern. This level should include any pattern that does not bind itself to a particular UI element. Styles written here can include both structural and cosmetic rules, and apply branding.
+They should be **reusable** and independent of context.
+
+They should be **modular**, which means they should have a single focus, and contain everything neccessary to display that part of the UI.
+
+They should be **isolated**, meaning they do not directly modify or depend on another component. This is more important than code reuse across components as this _could_ increase dependencies and tight coupling.
 
 #### Utilities
-Utilities are high-specificity, very explicit classes that apply a single rule or a single piece of functionality. They are used as overrides, helper classes, and to quickly build up simple page elements.
+Utilities are high-specificity, very explicit immutable classes that apply a single rule or a single piece of functionality. They are used as overrides, helper classes, and to quickly build up simple page elements.
```

Comment:
Maybe "quickly *scaffold* up simple page elements", to emphasise that building in this way should be more temporary and assists in building more solid components? 

Do we even want to come down on a side in the component vs atomic argument? (because our utility classes can easily be used as atomic units and I have View on this. I think using scoped components is 👍).

Reply:
I think we should avoid the use of the word "atomic" due to connotations with other CSS approaches.

Reply:
I think using the classes from utilities has significant use cases, but i think the way this is structured _should_ avoid atomic uses